File: ios/Classes/strategies/UnknownPermissionStrategy.m
//
// Created by Razvan Lung(long1eu) on 2019-02-15.
// Copyright (c) 2019 The Chromium Authors. All rights reserved.
//

#import "UnknownPermissionStrategy.h"


@implementation UnknownPermissionStrategy

- (PermissionStatus)checkPermissionStatus:(PermissionGroup)permission {
    return PermissionStatusUnknown;
}

- (ServiceStatus)checkServiceStatus:(PermissionGroup)permission {
    return ServiceStatusUnknown;
}

- (void)requestPermission:(PermissionGroup)permission completionHandler:(PermissionStatusHandler)completionHandler {
    completionHandler(PermissionStatusUnknown);
}
@end

File: ios/Classes/strategies/UnknownPermissionStrategy.swift
//
//  UnknownPermissionStrategy.swift
//  permission_handler
//
//  Created by Maurits van Beusekom on 07/08/2018.
//

import Foundation

class UnknownPermissionStrategy : NSObject, PermissionStrategy {
    func checkPermissionStatus(permission: PermissionGroup) -> PermissionStatus {
        return PermissionStatus.unknown
    }
    
    func checkServiceStatus(permission: PermissionGroup) -> ServiceStatus {
        return ServiceStatus.unknown
    }
    
    func requestPermission(permission: PermissionGroup, completionHandler: @escaping PermissionStatusHandler) {
        completionHandler(PermissionStatus.unknown)
    }
}
